exchanges/stream/websocket_test.go

```diff
@@ -1101,34 +1273,83 @@ func TestEnable(t *testing.T) {
 func TestSetupNewConnection(t *testing.T) {
 	t.Parallel()
 	var nonsenseWebsock *Websocket
-	err := nonsenseWebsock.SetupNewConnection(ConnectionSetup{URL: "urlstring"})
+	err := nonsenseWebsock.SetupNewConnection(&ConnectionSetup{URL: "urlstring"})
 	assert.ErrorIs(t, err, errWebsocketIsNil, "SetupNewConnection should error correctly")
 
 	nonsenseWebsock = &Websocket{}
-	err = nonsenseWebsock.SetupNewConnection(ConnectionSetup{URL: "urlstring"})
+	err = nonsenseWebsock.SetupNewConnection(&ConnectionSetup{URL: "urlstring"})
 	assert.ErrorIs(t, err, errExchangeConfigNameEmpty, "SetupNewConnection should error correctly")
 
 	nonsenseWebsock = &Websocket{exchangeName: "test"}
-	err = nonsenseWebsock.SetupNewConnection(ConnectionSetup{URL: "urlstring"})
+	err = nonsenseWebsock.SetupNewConnection(&ConnectionSetup{URL: "urlstring"})
 	assert.ErrorIs(t, err, errTrafficAlertNil, "SetupNewConnection should error correctly")
 
 	nonsenseWebsock.TrafficAlert = make(chan struct{}, 1)
-	err = nonsenseWebsock.SetupNewConnection(ConnectionSetup{URL: "urlstring"})
+	err = nonsenseWebsock.SetupNewConnection(&ConnectionSetup{URL: "urlstring"})
 	assert.ErrorIs(t, err, errReadMessageErrorsNil, "SetupNewConnection should error correctly")
 
 	web := NewWebsocket()
 
 	err = web.Setup(defaultSetup)
 	assert.NoError(t, err, "Setup should not error")
 
-	err = web.SetupNewConnection(ConnectionSetup{})
-	assert.ErrorIs(t, err, errExchangeConfigEmpty, "SetupNewConnection should error correctly")
-
-	err = web.SetupNewConnection(ConnectionSetup{URL: "urlstring"})
+	err = web.SetupNewConnection(&ConnectionSetup{URL: "urlstring"})
 	assert.NoError(t, err, "SetupNewConnection should not error")
 
-	err = web.SetupNewConnection(ConnectionSetup{URL: "urlstring", Authenticated: true})
+	err = web.SetupNewConnection(&ConnectionSetup{URL: "urlstring", Authenticated: true})
 	assert.NoError(t, err, "SetupNewConnection should not error")
+
+	// Test connection candidates for multi connection tracking.
+	multi := NewWebsocket()
+	set := *defaultSetup
+
+	// Values below are now not necessary as this will be set per connection
+	// candidate in SetupNewConnection.
```

Comment:
I don't understand this comment - If not necessary, why not remove?

Reply:
This was before when I was using the connector function as a pivot to use instead of the bool `UseMultiConnectionManagement` I will remove and the setting of nil below.